shared/chat/conversation/list/index.desktop.js

```diff
@@ -1,244 +1,269 @@
-    if ((this.props.selectedConversation !== prevProps.selectedConversation) ||
-        (this.state.messages !== prevState.messages)) {
-      this.state.isLockedToBottom && this._scrollToBottom()
+  componentWillReceiveProps (nextProps: Props) {
+    if (this.props.selectedConversation !== nextProps.selectedConversation ||
+      this.props.listScrollDownCounter !== nextProps.listScrollDownCounter) {
+      this.setState({isLockedToBottom: true})
     }
 
-    if (this.props.editLastMessageCounter !== prevProps.editLastMessageCounter) {
-      this.onEditLastMessage()
+    if (this.props.messageKeys.count() !== nextProps.messageKeys.count()) {
+      if (this.props.messageKeys.count() > 1 && this._lastRowIdx !== -1) {
+        const toFind = this.props.messageKeys.get(this._lastRowIdx)
+        this._keepIdxVisible = nextProps.messageKeys.indexOf(toFind)
+      }
     }
+  }
 
-    if (this.state.messages !== prevState.messages && prevState.messages.count() > 1) {
-      const prependedCount = this.state.messages.indexOf(prevState.messages.first())
-      const headerCount = this.props.headerMessages.count()
-      if (prependedCount !== -1) {
-        // Measure the new items so we can adjust our scrollTop so your position doesn't jump
-        const scrollTop = this.state.scrollTop + _.range(headerCount, headerCount + prependedCount)
-          .map(index => this._cellMeasurer.getRowHeight({index}))
-          .reduce((total, height) => total + height, 0)
-
-        // Disabling eslint as we normally don't want to call setState in a componentDidUpdate in case you infinitely re-render
-        this.setState({scrollTop}) // eslint-disable-line react/no-did-update-set-state
+  _updateBottomLock = (clientHeight: number, scrollHeight: number, scrollTop: number) => {
+    // meaningless otherwise
+    if (clientHeight) {
+      const isLockedToBottom = scrollTop + clientHeight >= scrollHeight - lockedToBottomSlop
+      if (this.state.isLockedToBottom !== isLockedToBottom) {
+        this.setState({isLockedToBottom})
       }
     }
   }
 
-  componentWillReceiveProps (nextProps: Props) {
-    if (this.props.selectedConversation !== nextProps.selectedConversation) {
-      this.setState({isLockedToBottom: true})
-      this._recomputeList()
+  _maybeLoadMoreMessages = debounce((clientHeight: number, scrollTop: number) => {
+    if (clientHeight && scrollTop === 0) {
+      this.props.onLoadMoreMessages()
     }
+  }, 500)
 
-    const willScrollDown = nextProps.listScrollDownCounter !== this.props.listScrollDownCounter
+  _onScroll = ({clientHeight, scrollHeight, scrollTop}) => {
+    this._scrollTop = scrollTop
+    this._updateBottomLock(clientHeight, scrollHeight, scrollTop)
+    this._maybeLoadMoreMessages(clientHeight, scrollTop)
+  }
 
-    if (!this.state.isScrolling || willScrollDown || this.state.isLockedToBottom) {
-      this._updateInternalMessages(nextProps)
+  _onResize = ({width}) => {
```

Comment:
this isn't debounced and doens't seem to need to be anymore. everything renders quickly